hudi-common/src/main/java/org/apache/hudi/common/table/timeline/MetadataConversionUtils.java

```diff
@@ -255,9 +254,6 @@ public static <T extends SpecificRecordBase> T convertCommitMetadata(HoodieCommi
     }
     hoodieCommitMetadata.getPartitionToWriteStats().remove(null);
     org.apache.hudi.avro.model.HoodieCommitMetadata avroMetaData = JsonUtils.getObjectMapper().convertValue(hoodieCommitMetadata, org.apache.hudi.avro.model.HoodieCommitMetadata.class);
-    if (hoodieCommitMetadata.getCompacted()) {
-      avroMetaData.setOperationType(WriteOperationType.COMPACT.name());
-    }
```

Comment:
So this line is redundant?

Reply:
@codope Can you help to check whether we need this explicit set-up for compaction action name?

Reply:
I think we need this line. Completed `compaction` action is `commit` but in the commit metadata file we set operation type as `COMPACT`. Then, during conflict resolution when we read a `commit` file, we look for the write operation type based on what is mentioned in the commit metadata file.

Reply:
@codope I had check the HoodieCommitMetadata where code is used to set compact true, the write operationType all can be set correct, so I think we not need check here. 
like this:
![image](https://github.com/user-attachments/assets/a82d8e84-36dc-4e23-b983-251546db1717)
And guess this is a problem leftover in history, a redundant logic.

Reply:
Did you check the reader for `HoodieCommitMetadata` where the `compacted` is set as true but the reader check the operation type instead.

Reply:
@danny0405 yes, compacted is not used anywhere but here, they only use `operation type`, so if ther `operation type` set correct, any op is right.

Reply:
Is the change related with the fix? If not just remove it and unblock this PR.

Reply:
yes, but need change the logical in ListingBasedRollbackStrategy for identify logcompact. Fortunately the complete instance logcompact is `deltacommit` compat is `commit`, I will change the logical

Reply:
are you saying  for log_compaction, the action name is also overwritten as "compact" ?

Reply:
@danny0405 no, for example 
- logcompat will product `001.logcompact.requested, 001.logcompact.inflight, 001.deltacommit`, action is `deltacommit` and it's `writeOprationType` will be overwrite from `logcompact` to `compact`.
- compat will product `001.compact.requested, 001.compact.inflight, 001.commit`, action is `commit` and it's `writeOprationType` will be overwrite from `compact` to `compact`.

Reply:
> logcompat will product 001.logcompact.requested, 001.logcompact.inflight, 001.deltacommit, action is deltacommit and it's writeOprationType will be overwrite from logcompact to compact

This is a bug I think, it seems the original fix is more reasonable, another question is in `[ListingBasedRollbackStrategy.java](https://github.com/apache/hudi/pull/11631/files#diff-365332eac63a078cd9009480fb3adc14feb755604a26bb07a19c1b0d9c8c8498)` line 125, if for log_compaction the "compacted" is set up as true, it looks like the action will also be overritten to "compaction" ?

Reply:
@danny0405 yeah, because log_compaction is `compacted`, so it's writeOperatorType will be changed to `compact` from `logcompact` and save this metadata in deltacommit instance. Now, if user rollback the deltacommit which is logcompact. In fact, this instance action is treated as `deltacommit`, but the metadata writeOperatorType is `compact` which save in this instance before, so the runtime action will be change from deltacommit to compact when rollback.

Reply:
let's move back to the original fix.

Reply:
ok